portality/scripts/manage_background_jobs.py

```diff
@@ -144,40 +170,311 @@ def query(self):
+                                   key=lambda x: x.schedule_time, reverse=True)
+    for r in unscheduled_huey_jobs[:example_size]:
+        print_huey_job_data(r)
+    print()
+
+    counter = Counter(r.bgjob_id for r in huey_rows if not r.is_scheduled)
+    counter = {k: v for k, v in counter.items() if v > 1}
+    if counter:
+        print(title(f'### Duplicated jobs ({total_counter(counter)})'))
+        for k, v in counter.items():
+            print(f'{k} {v}')
+
+    print_huey_jobs_counter(counter)
+
+    print(title('# DB records'))
+    print(title(f'### Processing'))
+    for j in find_bgjobs_by_status(constants.BGJOB_STATUS_PROCESSING):
+        print(job_to_str(j))
+    print()
+
+    bgjobs: List[BackgroundJob] = find_bgjobs_by_status(BGJOB_STATUS_QUEUED)
+    counter = Counter((j.action, j.status) for j in bgjobs)
+    print(title(f'### Queued ({sum(counter.values())})'))
+    for k, v in sorted(counter.items(), key=lambda x: x[0]):
+        print(f'{k[0]:30} {k[1]:10} {v}')
+    print()
+
+    print(title(f'### last {example_size} jobs'))
+    for j in sorted(bgjobs, key=lambda j: j.created_date, reverse=True)[:example_size]:
+        print(job_to_str(j))
+    print()
+
+    print(title('# Queued delta between DB and redis:'))
+    huey_only, db_only = find_huey_bgjob_delta(unscheduled_huey_jobs, bgjobs)
+    print_job_delta('### DB only', ((j.id, j.action) for j in db_only))
+    print_job_delta('### Redis only', ((i.bgjob_id, i.bgjob_action) for i in huey_only))
+
+
+def rm_all():
+    from portality.bll.services.huey_job import huey_job_service
+    if input(color_text.apply_color(
+            'WARNING: This will delete all jobs from redis and the database. Proceed? [y\\N] ',
+            background=color_text.Color.red)).lower() != 'y':
+        print('No action.')
+        return
+    from portality import models
+
+    print('Remove all jobs from DB')
+    models.BackgroundJob.delete_by_query(es_queries.query_all())
+
+    print('Remove all jobs from redis')
+    client = huey_job_service.create_redis_client()
+    client.delete('huey.redis.doajmainqueue')
+    client.delete('huey.redis.doajlongrunning')
+
+
+def rm_old_processing(is_all=False):
+    from portality import models
+    from portality.constants import BGJOB_STATUS_PROCESSING
+    from portality.models.background import BackgroundJobQueryBuilder
+
+    bgjobs = models.BackgroundJob.q2obj(q=BackgroundJobQueryBuilder()
+                                        .status_includes([BGJOB_STATUS_PROCESSING])
+                                        .order_by('created_date', 'asc')
+                                        .size(10000)
+                                        .build_query_dict())
+
+    if not is_all:
+        bgjobs = bgjobs[:-1]
+
+    if bgjobs:
+        print(f'Following {len(bgjobs)} jobs will be removed:')
+        for j in bgjobs:
+            print(job_to_str(j))
+        models.BackgroundJob.bulk_delete(b.id for b in bgjobs)
```

Comment:
Can I just confirm if this needs to also remove anything from huey?

Reply:
no needed, `rm_old_processing` used to remove processing backgroundjob in ES DB.
For the outer of huey or redis records will be removed in `rm_async_queued_jobs`
```
def cleanup():
    rm_redundant()
    rm_old_processing()

    time.sleep(5)  # wait DB records to be deleted
    rm_async_queued_jobs()
```